js/source/raster_tile_source.js

```diff
@@ -12,6 +13,15 @@ function RasterTileSource(options) {
     util.extend(this, util.pick(options, ['url', 'tileSize']));
 
     Source._loadTileJSON.call(this, options);
+    if (this.url) {
+        var onTokenChange = function() {
+            Source._loadTileJSON.call(this, util.extend({}, options, { preservePyramid: true }));
```

Comment:
You can use [`util.setOptions`](https://github.com/mapbox/mapbox-gl-js/blob/master/js/util/util.js#L371) in cases like this